src/de/arg/mod.rs

```diff
@@ -84,115 +83,102 @@ impl<'de> DeserializeSeed<'de> for ArgKV<'de> {
         formatter.write_str("Arg Map")
     }
 
     fn visit_map<A>(self, mut map: A) -> Result<Self::Value, A::Error>
     where
-        A: serde::de::MapAccess<'a>,
+        A: serde::de::MapAccess<'de>,
     {
         let mut arg = self.0;
 
         while let Some(key) = map.next_key::<&str>()? {
             arg = parse_value!(key, arg, map, Arg, {
                     // action : specailized
-                    (alias, &str),
-                    ref (aliases, Vec<&str>),
+                    (alias, String),
+                    ref (aliases, Vec<String>),
                     (allow_hyphen_values, bool),
-                    (allow_invalid_utf8, bool),
-                    (conflicts_with, &str),
-                    ref (conflicts_with_all, Vec<&str>),
-                    (default_missing_value, &str),
-                    ref (default_missing_values, Vec<&str>),
-                    // (default_missing_value_os, &OsStr), // need Deseriaze to OsStr
-                    // ref (default_missing_values_os, Vec<&OsStr>),
-                    (default_value, &str),
+                    (conflicts_with, String),
+                    ref (conflicts_with_all, Vec<String>),
+                    (default_missing_value, String),
+                    ref (default_missing_values, Vec<String>),
+                    (default_value, String),
                     // default_value_if : tuple3
-                    ref (default_value_ifs, Vec<(&str, Option<&str>, Option<&str>)> ),
+                    (default_value_ifs, Vec<(String, String, String)> ),
                     (display_order, usize),
                     // env : specialized
                     // env_os // not supported yet
                     (exclusive, bool),
-                    (forbid_empty_values, bool),
                     (global, bool),
-                    (group, &str),
-                    ref (groups, Vec<&str>),
-                    (help, &str),
-                    (help_heading, &str),
+                    (group, String),
+                    ref (groups, Vec<String>),
+                    (help, String),
+                    (help_heading, String),
                     (hide, bool),
                     (hide_default_value, bool),
                     // hide_env : specialized
                     // hide_env_values : specialized
                     (hide_long_help, bool),
                     (hide_possible_values, bool),
                     (hide_short_help, bool),
-                    (id, &str),
+                    (id, String),
                     (ignore_case, bool),
                     (index, usize),
                     (last, bool),
-                    (long, &str),
-                    (long_help, &str),
-                    (max_occurrences, usize),
-                    (max_values, usize),
-                    (min_values, usize),
-                    (multiple_occurrences, bool),
-                    (multiple_values, bool),
```

Comment:
> Add new keys 

`num_args` will be an interesting API to add.  Any thoughts on how you'd want to expose it?